src/app/shared/users/user/user.component.ts

```diff
@@ -0,0 +1,169 @@
+import {isNullOrUndefined} from 'util';
+
+@Component({
+  selector: 'user',
+  templateUrl: './user.component.html'
+})
+export class UserComponent implements OnInit, OnDestroy {
+
+  @Input() public resourceType: ResourceType;
+
+  @Input() public editable: boolean;
+
+  @Input() public changeRoles: boolean;
+
+  @Input() public user: UserModel;
+
+  @Input() public expanded: boolean;
+
+  @Input() public userRoles: string[];
+
+  @Input() public groupRoles: string[];
+
+  @Output() public expandedChange = new EventEmitter();
+
+  @Output() public userUpdated = new EventEmitter<UserModel>();
+
+  @Output() public userDeleted = new EventEmitter<UserModel>();
+
+  @Output() public rolesUpdate = new EventEmitter<{ roles: string[], onlyStore: boolean }>();
+
+  public showEmailWarning: boolean;
+
+  private lastSincedUserRoles: string[];
+  private rolesChange$ = new Subject<string[]>();
+  private rolesChangeSubscription: Subscription;
+
+  constructor(private i18n: I18n) {
+  }
+
+  public ngOnInit() {
+    this.rolesChangeSubscription = this.rolesChange$.pipe(
+      debounceTime(1000),
+      filter(newRoles => !deepArrayEquals(newRoles, this.lastSincedUserRoles))
+    ).subscribe(newRoles => {
+      this.lastSincedUserRoles = null;
+      this.rolesUpdate.emit({roles: newRoles, onlyStore: false});
+    });
+  }
+
+  public ngOnDestroy() {
+    if (this.rolesChangeSubscription) {
+      this.rolesChangeSubscription.unsubscribe();
+    }
+  }
+
+  public onNewEmail(email: string) {
+    if (!Validator.validateEmail(email)) {
+      this.showEmailWarning = true;
+      return;
+    }
+    this.userUpdated.emit({...this.user, email});
+  }
+
+  public onEmailFocus() {
+    this.showEmailWarning = false;
+  }
+
+  public removeUser() {
+    this.userDeleted.emit(this.user);
+  }
+
+  public getRoles(): Role[] {
+    switch (this.resourceType) {
+      case ResourceType.Organization:
+        return [Role.Read, Role.Manage, Role.Write];
+      case ResourceType.Project:
+        return [Role.Read, Role.Manage, Role.Write];
+      case ResourceType.Collection:
+        return [Role.Read, Role.Manage, Role.Write, Role.Share];
+      case ResourceType.View:
+        return [Role.Manage, Role.Clone, Role.Read];
+      default:
+        return [];
+    }
+  }
+
+  public getIconForRole(role: Role): string {
+    switch (role) {
+      case Role.Read:
+        return 'fa-book';
+      case Role.Manage:
+        return 'fa-cog';
+      case Role.Write:
+        return 'fa-pencil';
+      case Role.Clone:
+        return 'fa-clone';
+      case Role.Comment:
+        return 'fa-comment-alt';
+      case Role.Share:
+        return 'fa-share-square';
+      default:
+        return '';
+    }
+  }
+
+  public getTitleForRole(role: Role): string {
+    return this.i18n({
+      id: 'user.permission.icon',
+      value: '{role, select, READ {read} MANAGE {manage} WRITE {write} CLONE {clone} COMMENT {comment} SHARE {share}}'
```

Comment:
Does this ICU expression really work when you use it in element's attribute?

Reply:
Yes it works

Reply:
Perfect. Because Angular does not support it directly on the attributes yet. Using `i18n` service as you did is much better workaround than our previous solution.